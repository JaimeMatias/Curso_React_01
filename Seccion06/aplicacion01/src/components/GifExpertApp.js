import React, { useState } from 'react'
import { AddCategory } from './AddCategory';
import GifGrid from './GifGrid';
//import PropTypes from 'prop-types';


export const GifExpertApp = (initialData) => {
    //const categories = ['One Punch', 'Samurai X', 'Dragon Ball']
    const [categories, setCategories] = useState(initialData.initialData);


    return (
        <>
            <h2>GifExpertApp</h2>
           
            {// Para que exista la comunicación este componentes de distintos niveles, no lo que se puede hacer es crear en el componente hijo
                // un atribuco cuyo valor sea una referencia a la función de Hook del padre
                //En este caso con la función SetCategories. Así cada que en AddCategory se realiza un cambio, esto llega al GifExpertAdd
            }
            <AddCategory setCategories={setCategories} />
            
            <ol>
                {
                    // La función map, me permite iterar sobre los elementos de un arreglo cuando estoy en una pagina web
                    categories.map(category => 
                        <GifGrid 
                        key={category}
                        category={category}
                        />
                    )
                }
            </ol>

        </>
    )
}

//Sirve para definir los valores por defecto de las distintas propiedades del componente
GifExpertApp.defaultProps = {
    initialData: ['One Punch']
}


export default GifExpertApp;