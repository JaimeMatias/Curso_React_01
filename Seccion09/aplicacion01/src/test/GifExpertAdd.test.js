import React from "react";
import { shallow } from 'enzyme';
import { GifExpertApp } from "../GifExpertApp";

describe('Prueba <GifExpertAdd />', () => {
    //Generar el Snapshot
    
    test('debe mostrar el Componente correctamente ', () => {
        const wrapper = shallow(<GifExpertApp />);
        //Compara el componente generado con el snapshot que tenemos
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar una lista de categorias ', () => {
        
      const categories=['One Punch','Dragon Ball']
      const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('GifGrid').length).toBe(categories.length)
    })
    

})
