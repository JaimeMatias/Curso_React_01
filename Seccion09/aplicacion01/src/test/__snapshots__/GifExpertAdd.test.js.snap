// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba <GifExpertAdd /> debe de mostrar una lista de categorias  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Prueba <GifExpertAdd /> debe mostrar el Componente correctamente  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
